### PR TITLE
adding resource dms-endpoint

### PR DESCRIPTION
**Pull request by joshuaroot**
- per #1900
  - adding new resource `dms-endpoint`
  - adding resource action `set-ssl-mode`

### CODE REVIEW

**Comment by kapilt** [comment]
there's a ton of possible options on modify dms endpoint, rather than invoke each separately it would be nicer to have a generic action that can cover a greater functional set via pass through on options with schema validation.

**Comment by kapilt** [comment]
@joshuaroot exactly re unified modify action

File: c7n/resources/dms.py [hunk 1]
```diff
@@ -224,3 +238,199 @@ def process_resource_set(self, resources, tags):
...
+
+    def configure_s3_params(self, e, params):
+        settings = self.data.get('s3settings')
+        if not settings:
+            raise KeyError('s3settings not provided')
+
+        params['S3Settings'] = {
+            'BucketName': settings.get(
+                'bucketname', e['S3Settings']['BucketName']),
+            'CsvDelimiter': settings.get(
```
Comment (jhnlsn) [inline c7n/resources/dms.py]: in this case, is it ok that CsvDelimiter could end up None?

File: c7n/resources/dms.py [hunk 2]
```diff
@@ -224,3 +238,199 @@ def process_resource_set(self, resources, tags):
...
+                'csvdelimiter', e.get('S3Settings', {}).get(
+                    'CsvDelimiter')) or ',',
+            'CsvRowDelimiter': settings.get(
+                'csvrowdelimiter', e.get('S3Settings', {}).get(
+                    'CsvRowDelimiter')) or '\n',
+            'CompressionType': settings.get(
+                'compressiontype', e.get('S3Settings', {}).get(
+                    'CompressionType')) or 'none'}
```
Comment (jhnlsn) [inline c7n/resources/dms.py]: botodocs say default is NONE, is none (lowercase) the same?

Comment (joshuaroot) [reply c7n/resources/dms.py]: yes. NONE/ONE or none/one... case does not appear to be relevant but having said that, i might need to add a `lower()` to ensure it's uniform

**Comment by kapilt** [comment]
so since this is a passthrough, i'd rather we didn't do name mangling (lowercase) on the params so people can just ref the service docs.

**Comment by kapilt** [comment]
fixing this is probably a release blocker, i'd rather not have syntax in a release which we'll need to remove to cleanup.

**Pull request merged**